core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3016,6 +3016,50 @@ wasm_runtime_init_wasi(WASMModuleInstanceCommon *module_inst,
         fd_prestats_insert(prestats, dir_list[i], wasm_fd);
     }
 
+    for (i = 0; i < map_dir_count; i++, wasm_fd++) {
+        char mapping_copy[(PATH_MAX * 2) + 3];
```

Comment:
A concern is that the array size is very large (should be 4096 * 2 + 3), some platforms' stack may be not enough. Could we define a smaller array buffer and use it when it is enough, and allocate memory when needed? Like:
```C
    char mapping_copy_buf[256];
    char mapping_copy = mapping_copy_buf;

    if (strlen(map_dir_list) * 2 + 3 > 256) {
        if (!(mapping_copy = wasm_runtime_malloc(strlen(map_dir_list) * 2 + 3)))) {
            handle error;
        }
    }
    ...
    if (mapping_copy != mapping_copy_buf)
        wasm_runtime_free(mapping_copy);
```

Reply:
Will do

Reply:
Done.